onnxruntime/core/providers/cuda/rnn/cudnn_rnn_base.cc

```diff
@@ -348,14 +348,13 @@ Status CudnnRnnBase<T>::ComputeInternal(OpKernelContext* ctx) const {
   }
 
   if ((CUDNN_RNN_RELU == rnn_mode_ || CUDNN_RNN_TANH == rnn_mode_) && sequence_lens_data != nullptr && y_h_data != nullptr && y_data != nullptr) {
-    CudaAsyncBuffer<int32_t> sequence_lens_buffer(this, batch_size);
-    memcpy(sequence_lens_buffer.CpuPtr(), sequence_lens_data, batch_size * sizeof(int32_t));
-    ORT_RETURN_IF_ERROR(sequence_lens_buffer.CopyToGpu());
+    TArray<int32_t> sequence_lens_buffer(batch_size);
```

Comment:
>batch_size [](start = 41, length = 10)

not safe for batch_size > 8...I doubt if we have enough test case to catch this though.